code/datums/dna.dm

```diff
@@ -150,7 +150,9 @@ GLOBAL_LIST_INIT(total_uf_len_by_block, populate_total_uf_len_by_block())
 	SEND_SIGNAL(holder, COMSIG_CARBON_GAIN_MUTATION, mutation_type, class)
 	return force_give(new mutation_type (class, time, copymut = mutation))
 
-/datum/dna/proc/remove_mutation(mutation_type)
+/datum/dna/proc/remove_mutation(datum/mutation/human/mutation_type, mutadone)
+	if((mutation_type.mutadone_proof && mutadone))
```

Comment:
Shouldn't you be doing this check after get_mutation()?

Reply:
mutation_type is a type, not an instanced object. But even if you used initial here to grab the type's value, you're checking the value of the mutation you pass in and not what's actually in the dna.

Reply:
guh